app/controllers/flowchart_controller.rb

```diff
@@ -1,4 +1,103 @@
 # frozen_string_literal: true
 
 class FlowchartController < ApplicationController
+  def create
+    @flowchart = Flowchart.new(JSON.parse(request.body.read))
+
+    unless @flowchart.valid?
+      render status: 400, json: { error: 'Invalid Flowchart params' }
+      return
+    end
+
+    @flowchart = Flowchart.create(JSON.parse(request.body.read))
+    root_node = FlowchartNode.create(flowchart_id: @flowchart.id, text: 'New Node', header: 'Options', is_root: true, deleted: false)
+    @flowchart[:root_id] = root_node.id
+    @flowchart.save
+
+    render json: @flowchart
+  end
+
+  def update
+    flowchart = Flowchart.find(params[:id])
+    if flowchart.nil?
+      render status: 404, json: { error: 'Could not find flowchart' }
+    end
+
+    flowchart.update_attributes(JSON.parse(request.body.read))
+    unless flowchart.valid?
+      render status: 400, json: { error: 'Invalid flowchart params' }
+      return
+    end
+
+    @flowchart = Flowchart.update(params[:id], JSON.parse(request.body.read))
+    render json: @flowchart
+  end
+
+  def delete
+    @flowchart = Flowchart.find(params[:id])
+    if @flowchart.nil?
+      render status: 404, json: { error: 'Could not find flowchart' }
+    end
+
+    Flowchart.find(params[:id]).update(deleted: true)
+    FlowchartNode.where(flowchart_id: params[:id]).update_all(deleted: true)
+    @flowchart[:deleted] = true
+
+    render json: @flowchart
+  end
+
+  def all_flowcharts
+    render json: Flowchart.where(deleted: false)
+  end
+
+  def serialized_flowchart_by_id
+    flowchartnodes = FlowchartNode.where(flowchart_id: params[:id], deleted: false)
+    flowchart = Flowchart.find_by(id: params[:id], deleted: false)
+    root_node = FlowchartNode.get_root_node(params[:id])
```

Comment:
Since `root_id` is an attribute of Flowchart, could we call `FlowchartNode.find(flowchart.root_id)` here?